ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/LfValueTranslation.scala

```diff
@@ -0,0 +1,222 @@
+      "exercise_argument" -> serializeExerciseArgOrThrow(exercise),
+      "exercise_result" -> serializeNullableExerciseResultOrThrow(exercise),
+    )
+  }
+
+  private def toApiValue(
+      value: LfValue,
+      verbose: Boolean,
+      attribute: => String,
+  ): ApiValue =
+    LfEngineToApi.assertOrRuntimeEx(
+      failureContext = s"attempting to deserialize persisted $attribute to value",
+      LfEngineToApi
+        .lfVersionedValueToApiValue(
+          verbose = verbose,
+          value = value,
+        ),
+    )
+
+  private def toApiRecord(
+      value: LfValue,
+      verbose: Boolean,
+      attribute: => String,
+  ): ApiRecord =
+    LfEngineToApi.assertOrRuntimeEx(
+      failureContext = s"attempting to deserialize persisted $attribute to record",
+      LfEngineToApi
+        .lfVersionedValueToApiRecord(
+          verbose = verbose,
+          recordValue = value,
+        ),
+    )
+
+  def deserialize[E](raw: Raw.Created[E], verbose: Boolean): CreatedEvent = {
+    val key = LfValueTranslation.Cache.Key(raw.partial.eventId)
+    val create =
+      cache
+        .getIfPresent(key = key)
+        .getOrElse(
+          LfValueTranslation.Cache.Value.Create(
+            argument = ValueSerializer.deserializeValue(raw.createArgument),
+            key = raw.createKeyValue.map(ValueSerializer.deserializeValue)
+          )
+        )
+        .assertCreate()
+    raw.partial.copy(
+      createArguments = Some(
+        toApiRecord(
+          value = create.argument,
+          verbose = verbose,
+          attribute = "create argument",
+        )
+      ),
+      contractKey = create.key.map(
+        key =>
+          toApiValue(
+            value = key,
+            verbose = verbose,
+            attribute = "create key",
+        )
+      ),
+    )
+  }
+
+  def deserialize(raw: Raw.TreeEvent.Exercised, verbose: Boolean): ExercisedEvent = {
+    val key = LfValueTranslation.Cache.Key(raw.partial.eventId)
+    val exercise =
+      cache
+        .getIfPresent(key = key)
+        .getOrElse(
+          LfValueTranslation.Cache.Value.Exercise(
+            argument = ValueSerializer.deserializeValue(raw.exerciseArgument),
+            result = raw.exerciseResult.map(ValueSerializer.deserializeValue)
+          )
+        )
+        .assertExercise()
+    raw.partial.copy(
+      choiceArgument = Some(
+        toApiValue(
+          value = exercise.argument,
+          verbose = verbose,
+          attribute = "exercise argument",
+        )
+      ),
+      exerciseResult = exercise.result.map(
+        result =>
+          toApiValue(
+            value = result,
+            verbose = verbose,
+            attribute = "exercise result",
+        )
+      ),
+    )
+  }
+
+}
+
+object LfValueTranslation {
+
+  type Cache = caching.Cache[Cache.Key, Cache.Value]
+
+  object Cache {
+
+    // To avoid the overhead of measuring the size of objects, we just assign a weight of 1 to values
```

Comment:
Probably worth explaining why the weight here is 0. I had to read the comment three times before I understood.

Reply:
Is there any way to change the interface to `Cache` so this wouldn't create a weighted cache? I expect the performance of Caffeine would be better if we didn't use weighting at all in this situation, and instead used size-based eviction.

Reply:
I played around with it a bit but it's starting to play around with implicits and adding too many methods, I'd prefer simply improving the comment right now and leave this for another PR.

Reply:
Created https://github.com/digital-asset/daml/issues/6057

Reply:
Comment improved in https://github.com/digital-asset/daml/pull/6052/commits/c99211fe5b22dad2ebd66ee97263517279f3f005